pkg/network/render.go

```diff
@@ -886,6 +908,110 @@ func renderIPTablesAlerter(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.
 	return manifests, nil
 }
 
+// renderNetworkingConsolePlugin renders the common objects related to the networking console plugin resources
+func renderNetworkingConsolePlugin(manifestDir string, cl cnoclient.Client) ([]*uns.Unstructured, error) {
+	if !isConsolePluginCRDExist(cl) {
+		log.Printf("consoleplugins.console.openshift.io CRD does not exist yet")
+		return nil, nil
+	}
+
+	tmpFallbackConsoleImage := "quay.io/orenc/networking-console-plugin:4.16"
+	data := render.MakeRenderData()
+	data.Data["ReleaseVersion"] = os.Getenv("RELEASE_VERSION")
+
+	val, ok := os.LookupEnv("NETWORKING_CONSOLE_PLUGIN_IMAGE")
+	if ok {
+		data.Data["NetworkingConsolePluginImage"] = val
+	} else {
+		data.Data["NetworkingConsolePluginImage"] = tmpFallbackConsoleImage
+	}
+
+	manifests, err := render.RenderDir(filepath.Join(manifestDir, "networking-console-plugin"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render networking-console-plugin manifests")
+	}
+	return manifests, nil
+}
+
+type Patch struct {
+	Type types.PatchType
+	Data []byte
+}
+
+// registerNetworkingConsolePlugin enables console plugin for networking-console if not already enabled
+func registerNetworkingConsolePlugin(cl cnoclient.Client) error {
```

Comment:
What will happen when the console plugin is not deployed and we register the plugin?
A more general question: why doesn't the console just consume the plugins that were created?

Reply:
>What will happen when the console plugin is not deployed and we register the plugin?
nothing disastrous, the console will just drop some errors the the plugin could not be loaded in its logs and in the browser's console.

> A more general question: why doesn't the console just consume the plugins that were created?
because in the general case of a regular operator with a custom console plugin, the admin should have an option to disable it if s/he wants.
in our case, we're auto-registering it because it should be always enabled, as the networking section is a core section in the UI.